File: LeetCode/Python Solutions/Redistribute Characters to Make All Strings Equal/redistribute.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        from collections import Counter

        # Count all characters in all words
        total_count = Counter(''.join(words))

        # Check if each character's total count is divisible by the number of words
        for char, count in total_count.items():
            if count % len(words) != 0:
                return False

        return True

File: LeetCode/Java Solutions/Redistribute Characters to Make All Strings Equal/redistribute.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean makeEqual(String[] words) {
        Map<Character, Integer> charCount = new HashMap<>();

        // Count all characters in all words
        for (String word : words) {
            for (char c : word.toCharArray()) {
                charCount.put(c, charCount.getOrDefault(c, 0) + 1);
            }
        }

        // Check if each character's total count is divisible by the number of words
        for (int count : charCount.values()) {
            if (count % words.length != 0) {
                return false;
            }
        }

        return true;
    }
}
